pkg/chunk/encoding/chunk.go

```diff
@@ -101,6 +101,40 @@ type Batch struct {
 	Length     int
 }
 
+func (b *Batch) dropValuesForInterval(interval model.Interval) {
+	if b.Length == 0 || model.Time(b.Timestamps[0]) > interval.End || model.Time(b.Timestamps[b.Length-1]) < interval.Start {
+		return
+	}
+
+	removedIdxStart := 0
```

Comment:
I am not sure being super performant is important here. Can we simplify this by making a new array and iterating through the samples?

like 
```
reducedTs := [BatchSize]int64
reducedVals := [BatchSize]float64
length := 0

for i := range b.Timestamps {
  if inbound(model.Time(b.Timestamps[i], interval) {
    continue
  }
  reducedTs[length] = b.Timestamps[i]
  reducedValues[length] = b.Values[i]
}

b.Timestamps = reducedTs
b.Values = reducedVals
b.Length = length
```